builtin/providers/aws/resource_aws_kinesis_analytics_test.go

```diff
@@ -0,0 +1,287 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/kinesisanalytics"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSKinesisAnalytics_basic(t *testing.T) {
+	var desc kinesisanalytics.ApplicationDetail
+
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: destroyKinesisAnalytics,
+		Steps: []resource.TestStep{
+			{
+				Config: kinesisAnalyticsBasicCreateConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					doesKinesisAnalyticsInstanceExist("aws_kinesis_analytics.test_application", &desc),
+					areRootAttributesCorrect(&desc),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSKinesisAnalytics_stream_connections(t *testing.T) {
+	var desc kinesisanalytics.ApplicationDetail
+
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: destroyKinesisAnalytics,
+		Steps: []resource.TestStep{
+			{
+				Config: kinesisAnalyticsCreateWithStreamsConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					doesKinesisAnalyticsInstanceExist("aws_kinesis_analytics.test_application", &desc),
+					areInputStreamAttributesCorrect(&desc),
+					areOutputStreamAttributesCorrect(&desc),
+				),
+			},
+		},
+	})
+}
+
+func doesKinesisAnalyticsInstanceExist(n string, desc *kinesisanalytics.ApplicationDetail) resource.TestCheckFunc {
```

Comment:
Take a look at some of the other test files to see a naming convention for this - starting a func with `does` doesn't follow that convention